dig.go

```diff
@@ -213,21 +213,38 @@ type VisualizeOption interface {
 	unimplemented()
 }
 
-var _graphTmpl = template.Must(template.New("DotGraph").Parse(`digraph {
+var funcMap = template.FuncMap{
+	"quote": quote,
+}
+
+var _graphTmpl = template.Must(template.New("DotGraph").Funcs(funcMap).Parse(`digraph {
 	graph [compound=true];
+	{{range $g := .Groups}}
+		{{- quote .String}} [shape=diamond label=<{{.Type}}{{.Attributes}}>];
+		{{range .Results}}
+			{{- quote $g.String}} -> {{quote .String}};
+		{{end}}
+	{{end -}}
 	{{range $index, $ctor := .Ctors}}
 		subgraph cluster_{{$index}} {
-			{{printf "%q" .Name}} [shape=plaintext];
-			{{range $res := .Results}}
-				{{printf "%q" .String}} [label=<{{.Type}}{{.Attributes}}>];
+			{{quote .Name}} [shape=plaintext];
+			{{range .Results}}
+				{{- quote .String}} [label=<{{.Type}}{{.Attributes}}>];
 			{{end}}
 		}
-		{{range $par := .Params}}
-			{{printf "%q" $ctor.Name}} -> {{printf "%q" .String}} [ltail=cluster_{{$index}}{{if .Optional}} style=dashed{{end}}];
+		{{range .Params}}
+			{{- quote $ctor.Name}} -> {{quote .String}} [ltail=cluster_{{$index}}{{if .Optional}} style=dashed{{end}}];
 		{{end}}
+		{{range .GroupParams}}
+			{{- quote $ctor.Name}} -> {{quote .String}} [ltail=cluster_{{$index}}];
+		{{end -}}
 	{{end}}
 }`))
 
+func quote(s string) string {
+	return strconv.Quote(s)
```

Comment:
nit: You don't need to declare this function. You can do, `"quote": strconv.Quote`.